PhotoMap/Features/Map/ViewModels/MapViewModel.swift

```diff
@@ -85,10 +85,41 @@ class MapViewModel: MapViewModelType {
             .map { _ in () }
             .subscribe(coordinator.disableLocationSubject)
             .store(in: cancelBag)
+
+        coordinator.imagePickerSourceSubject
+            .sink(receiveValue: { [weak self] source in
+                guard let self = self else { return }
+
+                self.coordinator.showImagePickerSubject.send(self.createImagePicker(from: source))
+            })
+            .store(in: cancelBag)
     }
 
     private func switchFollowDiscoveryMode(disableFolowMode: Bool = false) {
         isFollowModeOn = disableFolowMode ? !disableFolowMode : !isFollowModeOn
         modeButtonCollor = isFollowModeOn ? Asset.followModeColor.color : Asset.discoverModeColor.color
     }
+
+    private func createImagePicker(from source: UIImagePickerController.SourceType) -> UIImagePickerController {
+        let picker = UIImagePickerController()
+        picker.sourceType = source
+        picker.delegate = self
+
+        return picker
+    }
+}
+
+extension MapViewModel: UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+    func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
+        picker.dismiss(animated: true, completion: nil)
+    }
+
+    func imagePickerController(_ picker: UIImagePickerController,
+                               didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
+        picker.dismiss(animated: true, completion: nil)
+
+        guard let image = info[UIImagePickerController.InfoKey.originalImage] as? UIImage else { return }
+        
+        coordinator.showMapPopupSubject.send(Photo(image: image))
+    }
```

Comment:
This looks like another separated coordinator, isn't it?

Reply:
Create separated coordinator "ImagePickerCoordinator" 6b97fdb